src/domain/siwe/siwe.repository.interface.ts

```diff
@@ -0,0 +1,24 @@
+import { AuthApiModule } from '@/datasources/auth-api/auth-api.module';
+import { SiweRepository } from '@/domain/siwe/siwe.repository';
+import { VerifyAuthMessageDto } from '@/routes/auth/entities/verify-auth-message.dto.entity';
+import { Module } from '@nestjs/common';
+
+export const ISiweRepository = Symbol('ISiweRepository');
+
+export interface ISiweRepository {
+  generateNonce(): Promise<{ nonce: string }>;
+
+  isValidMessage(args: VerifyAuthMessageDto): Promise<boolean>;
+}
+
+@Module({
+  imports: [AuthApiModule],
```

Comment:
Maybe we should rename the `AuthApiModule` as it's now SIWE related only 🤔

Reply:
I renamed it (and all other SiWe-associated datasource elements) to use `Siwe*` in place of `Auth*` in b15fa98e39c07d2987f9f21eafce4ea6e318ec21.